### DISCUSSION TITLE
Guest account without autologin and with password?

### DISCUSSION THREAD

**Discussion by JB-Prof** [ltsp/ltsp#298] [Q&A]
Hello,

We are looking to create about 20 accounts for generic use.  The login name would be student1 (or guest1 or something) through to student20.  The plan would be that the password would be the same as the username, and when they log out everything gets reset to the way it was before.

I've seen posts that describe how to set up a guest account with auto-login (such as a Kiosk) or a guest account without a password, but is a guide for how to set up this specific setup please?

Thank you.

**Comment by alkisg** [comment]
"Guest accounts with passwords" are normal accounts, no action is needed for that.
"Reset the home directory on login or logout" isn't specific to LTSP, as LTSP doesn't provide any hooks for login or logout (yet).

Three ways to get a login or logout hook are:

1. Override the desktop environment session
2. Force a server-side command on SSH/SFTP connections
3. Execute a command on server logins (as SSH/SFTP are) with pam_exec

I'll give a small how-to for "1" below, but as these things are not LTSP specific, I can't spend a lot of time documenting them; it's up to users like you to create guides in wiki pages and ask for a bit of help wherever needed.
If a user is willing to create a wiki page with HIS setup AND maintain it, I'm also willing to try to help him; as more users will get help that way, rather than answering to a separate discussion for each user.

The MATE session is provided by /usr/bin/mate-session. You can create an /etc/ltsp/mate-session shell script, with the following contents:

```shell
#!/bin/sh

case "$(id -un)" in
    student*)
        xterm
        ;;
esac
exec /usr/bin/mate-session "$@"
```

Then you can override /usr/bin/mate-session with it, by symlinking it to /usr/local/bin, which has bigger priority, from ltsp.conf:

```shell
[clients]
POST_INIT_SYMLINK_SESSION="ln -s /etc/ltsp/mate-session /usr/local/bin/"
```

These instructions will open an xterm every time a student tries to login. You're supposed to find the correct rm / copy commands to reset home, and replace the xterm command with them.
